include/crud-terms.php

```diff
@@ -293,54 +293,21 @@ public function set_pre_term_name( $name ) {
 	 * @return string Slug with a language suffix if found.
 	 */
 	public function set_pre_term_slug( $slug, $taxonomy ) {
-		if ( ! $this->model->is_translated_taxonomy( $taxonomy ) ) {
-			return $slug;
-		}
-
-		if ( ! $slug ) {
-			$slug = sanitize_title( $this->pre_term_name );
-		}
-
-		if ( ! term_exists( $slug, $taxonomy ) ) {
-			return $slug;
-		}
-
-		/**
-		 * Filters the subsequently inserted term language.
-		 *
-		 * @since 3.3
-		 *
-		 * @param PLL_Language|null $lang     Found language object, null otherwise.
-		 * @param string            $taxonomy Term taonomy.
-		 * @param string            $slug     Term slug
-		 */
-		$lang = apply_filters( 'pll_inserted_term_language', null, $taxonomy, $slug );
-
-		if ( ! $lang instanceof PLL_Language ) {
-			return $slug;
+		$term_slug = new PLL_Term_Slug( $this->model );
+		$term_slug->set_slug( $slug );
+		$term_slug->set_taxonomy( $taxonomy );
+		$term_slug->set_name( $this->pre_term_name );
```

Comment:
@Hug0-Drelon Would you prefer a global setter like `set_term_data()` that sets everything at once, or each item one by one like here?

Reply:
What about passing all to thé constructor?